lib/character/codes.d.ts

```diff
@@ -0,0 +1,140 @@
+// This module is generated by `script/`.
```

Comment:
since we currently don't have a "compile" step for `.d.ts` (we could have "copy" step for them), I'm not sure what do you mean by "compiled away".

Reply:
These constants are inlined: https://github.com/micromark/micromark/blob/main/script/babel-transform-constants.js. They’re not used in the code in `dist/`

Reply:
I get your idea. Since the `.d.ts` files for all compiled away files are generated (see `script/generate-constant-typings.js`), I can generate something like this 

```ts
// lib/character/codes.d.ts
// This module is generated by `script/`.

export type Code = -5 | -4 | -3

// @script: REMOVE_ALL_THING_BELOW

export interface Codes {
  "carriageReturn": -5,
  "lineFeed": -4,
  "carriageReturnLineFeed": -3,
}

declare const value: Codes
export default value

// @script: REMOVE_ALL_THING_ABOVE
```

and remove all content between two comments when coping `*.d.ts` from `lib/` to `dist/`. By doing that, `dist/character/codes.d.ts` will only export a *type* without any *value*. 

There is several ways to do that, but this way is the fastest to implement.

Reply:
update: implemented.

Reply:
So `Code` is simulated to `Type`: a value which is not defined in the `micromark` repository can still be valid. 

Does other `Code` can also from an extra plugin like `micromark-extension-gfm`?

Reply:
character codes come from strings: `'🤔'.charCodeAt(n)`